cmd/analyze/main.go

```diff
@@ -23,6 +24,13 @@ import (
 	"github.com/ossf/package-analysis/pkg/api/pkgecosystem"
 )
 
+const (
```

Comment:
The more idiomatic way to do this in Go is to return an error and check for its type to determine the kind of error. In this case, you can declare a `usageError` type and then return that, check for it in main, and return exit code 2 in that case. (for all other errors just return error code 1)

```
type usageError struct {
  error
}

func usagef(format string, args ...any) error {
  return usageError{fmt.Errorf(format, args...)}
}

func run() error {
  ...
  if err := featureflags.Update(*features); err != nil {
    return usagef("failed to parse flags: %w", err)
  }
  ...
  return nil
}

func main() {
  if err := run(); err !=  nil {
    fmt.Fprintln(os.Stderr, err)
    if errors.As(err, &usageError{}) {
      os.Exit(2)
    }
    os.Exit(1)
  }
}
```

Reply:
Thanks for the suggestion, this idiom is really nice! I'll implement it.